metabase/metabase
# Resizable Pivot Table Data Columns

resolves https://github.com/metabase/metabase/issues/27691

## Description

Allow users to resize data/value columns manually. We do not even try to do any automated measuring of columns because there is a potentially enormous number of them.

## Screenshots

![resizeDataCols](https://user-images.githubusercontent.com/30528226/212987058-44f0d22a-a2e8-48ef-9d50-af7724ccf7c1.gif)

![rearrange dimensions](https://user-images.githubusercontent.com/30528226/212987728-6cc45762-6132-435d-a65d-4c9f26288a8d.gif)

## Testing Steps

Open an enormous, gross pivot table like this one:
```
http://localhost:3000/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjoxLCJxdWVyeSI6eyJzb3VyY2UtdGFibGUiOjIsImFnZ3JlZ2F0aW9uIjpbWyJtYXgiLFsiZmllbGQiLDEzLG51bGxdXSxbImNvdW50Il0sWyJhdmciLFsiZmllbGQiLDEzLG51bGxdXV0sImJyZWFrb3V0IjpbWyJmaWVsZCIsMTQseyJ0ZW1wb3JhbC11bml0IjoicXVhcnRlciJ9XSxbImZpZWxkIiw4LHsic291cmNlLWZpZWxkIjo5fV0sWyJmaWVsZCIsMTIseyJiaW5uaW5nIjp7InN0cmF0ZWd5IjoiZGVmYXVsdCJ9fV0sWyJmaWVsZCIsMTcseyJiaW5uaW5nIjp7InN0cmF0ZWd5IjoiZGVmYXVsdCJ9fV1dfSwidHlwZSI6InF1ZXJ5In0sImRpc3BsYXkiOiJwaXZvdCIsImRpc3BsYXlJc0xvY2tlZCI6dHJ1ZSwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnsicGl2b3RfdGFibGUuY29sdW1uX3NwbGl0Ijp7InJvd3MiOltbImZpZWxkIiwxNCx7InRlbXBvcmFsLXVuaXQiOiJxdWFydGVyIn1dXSwiY29sdW1ucyI6W1siZmllbGQiLDEyLHsiYmlubmluZyI6eyJzdHJhdGVneSI6Im51bS1iaW5zIiwibWluLXZhbHVlIjowLCJtYXgtdmFsdWUiOjEwMCwibnVtLWJpbnMiOjgsImJpbi13aWR0aCI6MTIuNX19XSxbImZpZWxkIiw4LHsic291cmNlLWZpZWxkIjo5fV0sWyJmaWVsZCIsMTcseyJiaW5uaW5nIjp7InN0cmF0ZWd5IjoibnVtLWJpbnMiLCJtaW4tdmFsdWUiOjAsIm1heC12YWx1ZSI6NzAsIm51bS1iaW5zIjo4LCJiaW4td2lkdGgiOjEwfX1dXSwidmFsdWVzIjpbWyJhZ2dyZWdhdGlvbiIsMV0sWyJhZ2dyZWdhdGlvbiIsMl0sWyJhZ2dyZWdhdGlvbiIsMF1dfX0sIm9yaWdpbmFsX2NhcmRfaWQiOjIxfQ==
```
- see that you can resize any column (when there are multiple column headers, only the bottom one should be resizable)
- see that rearranging field dimensions in viz settings works properly, and resets column widths

## Remaining work to be done:
Persist column sizes: https://github.com/metabase/metabase/issues/27598
